src/entry_point.py

```diff
@@ -201,6 +256,314 @@ def _constructor_menuinst(prefix, pkg_names=None, root_prefix=None, remove=False
+            if target_path.startswith("HKEY"):
+                # target_path for cmd.exe is a registry path
+                reg_entry, _ = _read_windows_registry(target_path)
+                if not isinstance(reg_entry, str):
+                    continue
+                autorun_parts = reg_entry.split("&")
+                for prefix in prefixes:
+                    hook = str(prefix / "condabin" / "conda_hook.bat")
+                    if any(hook in part for part in autorun_parts):
+                        reverse_plan.append(initializer)
+                        break
+            else:
+                target_path = Path(target_path)
+                # Only reverse for paths that are outside the uninstall prefix
+                # since paths inside the uninstall prefix will be deleted anyway
+                if not target_path.exists() or _is_subdir(
+                    target_path, uninstall_prefix
+                ):
+                    continue
+                rc_content = target_path.read_text()
+                if shell == "powershell":
+                    pattern = CONDA_INITIALIZE_PS_RE_BLOCK
+                else:
+                    pattern = CONDA_INITIALIZE_RE_BLOCK
+                flags = re.MULTILINE
+                matches = re.findall(pattern, rc_content, flags=flags)
+                if not matches:
+                    continue
+                for prefix in prefixes:
+                    # Ignore .exe suffix to make the logic simpler
+                    if shell in ("csh", "tcsh") and sys.platform != "win32":
+                        sentinel_str = str(prefix / "etc" / "profile.d" / "conda.csh")
+                    else:
+                        sentinel_str = str(prefix / BIN_DIRECTORY / "conda")
+                    if sys.platform == "win32" and shell != "powershell":
+                        # Remove /cygdrive to make the path shell-independent
+                        sentinel_str = win_path_to_unix(sentinel_str).removeprefix(
+                            "/cygdrive"
+                        )
+                    if any(sentinel_str in match for match in matches):
+                        reverse_plan.append(initializer)
+                        break
+    return reverse_plan
+
+
+def _constructor_uninstall_subcommand(
+    uninstall_dir: str,
+    conda_clean: bool = False,
+    remove_condarcs: str | None = None,
+    remove_caches: bool = False,
+):
+    """
+    Remove a conda prefix or a directory containing conda environments.
+
+    This command also provides options to remove various cache and configuration
+    files to fully remove a conda installation.
+    """
+    from conda.base.constants import PREFIX_MAGIC_FILE
+
+    # See: https://github.com/conda/conda/blob/475e6acbdc98122fcbef4733eb8cb8689324c1c8/conda/gateways/disk/create.py#L482-L488 # noqa
+    ENVS_DIR_MAGIC_FILE = ".conda_envs_dir_test"
+
+    uninstall_prefix = Path(uninstall_dir).expanduser().resolve()
```

Comment:
Do we need to `os.path.expandvars` too or is that not expected?

Reply:
Good call. I added it to where we parse `args.prefix`.